src/LykkePublicAPI/Controllers/CandlesController.cs

```diff
@@ -19,7 +21,37 @@ public CandlesController(ICandlesHistoryServiceProvider candlesServiceProvider)
         }
 
         /// <summary>
-        /// Get candles for specified period and asset pair
+        /// Get list of supported asset pairs for the given market type.
+        /// </summary>
+        /// <param name="market">The market type. Acceptable values: Spot, Mt.</param>
+        [HttpGet("history/{market}/available")]
+        [ProducesResponseType(typeof(string[]), 200)]
+        [ProducesResponseType(typeof(ApiError), 400)]
+        public async Task<IActionResult> GetAvailableAssets(MarketType market)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState.ToApiError());
+            }
+
+            var candlesService = _candlesServiceProvider.TryGet(market.ToDomain());
+
+            if (candlesService == null)
+            {
+                return BadRequest(new ApiError
+                {
+                    Code = ErrorCodes.InvalidInput,
+                    Msg = "Invalid market"
+                });
+            }
+
+            var response = await candlesService.GetAvailableAssetPairsAsync();
```

Comment:
Can we get here any failed results? Is result check required?

Reply:
I think, we sould not check it additionally here. Actually, the underlying service just returns HTTP 200 and empty data in case of any faulty situation, and we translate it to client.